README.rst

```diff
@@ -13,1837 +13,30 @@ Turn complex GraphQL queries into optimized database queries.
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-
-The :code:`has_edge_degree` filter allows filtering based on the number of
-edges of a particular type. There are situations in which filtering with
-:code:`has_edge_degree` and filtering using :code:`=` on :code:`_x_count` produce
-equivalent queries. Here is one such pair of queries:
-
-.. code::
-
-    {
-        Species {
-            name @output(out_name: "name")
-            in_Animal_OfSpecies @filter(op_name: "has_edge_degree", value: ["$num_animals"]) {
-                uuid
-            }
-        }
-    }
-
-and
-
-.. code::
-
-    {
-        Species {
-            name @output(out_name: "name")
-            in_Animal_OfSpecies @fold {
-                _x_count @filter(op_name: "=", value: ["$num_animals"])
-            }
-        }
-    }
-
-In both of these queries, we ask for the names of the :code:`Species`
-vertices that have precisely :code:`$num_animals` members. However, we have
-expressed this question in two different ways: once as a property of the
-:code:`Species` vertex ("the degree of the :code:`in_Animal_OfSpecies` is
-:code:`$num_animals`"), and once as a property of the list of :code:`Animal`
-vertices produced by the :code:`@fold` ("the number of elements in the
-:code:`@fold` is :code:`$num_animals`").
-
-When we add additional filtering within the :code:`Animal` vertices of the
-:code:`in_Animal_OfSpecies` vertex field, this distinction becomes very
-important. Compare the following two queries:
-
-.. code::
-
-    {
-        Species {
-            name @output(out_name: "name")
-            in_Animal_OfSpecies @filter(op_name: "has_edge_degree", value: ["$num_animals"]) {
-                out_Animal_LivesIn {
-                    name @filter(op_name: "=", value: ["$location"])
-                }
-            }
-        }
-    }
-
-versus
-
-.. code::
-
-    {
-        Species {
-            name @output(out_name: "name")
-            in_Animal_OfSpecies @fold {
-                out_Animal_LivesIn {
-                    _x_count @filter(op_name: "=", value: ["$num_animals"])
-                    name @filter(op_name: "=", value: ["$location"])
-                }
-            }
-        }
-    }
-
-In the first, for the purposes of the :code:`has_edge_degree` filtering, the
-location where the animals live is irrelevant: the :code:`has_edge_degree`
-only makes sure that the :code:`Species` vertex has the correct number of
-edges of type :code:`in_Animal_OfSpecies`, and that's it. In contrast, the
-second query ensures that only :code:`Species` vertices that have
-:code:`$num_animals` animals that live in the selected location are returned
--- the location matters since the :code:`@filter` on the :code:`_x_count` field
-applies to the number of elements in the :code:`@fold` scope.
-
-The GraphQL schema
-------------------
```

Comment:
I did not include this section in ReadTheDocs. I included a more general back-end agnostic schema types page to go over the specific GraphQL compiler directives and how GraphQL concepts such as objects map to tables/vertex types in the underlying database.

This section could in the OrientDB ReadTheDocs page to help users manually construct a OrientDB schema and give some intuition to the design decisions you need to make when manually generating one. That being said, we now have a schema reflection function so I don't see that much value in this section. Thoughts?

Reply:
Yeah, this section talks about two things: What a typical schema looks like, and what is the result from orientdb schema generation. The former is already explained in the "Language Specification -> Schema Types" section, and the latter is too orientdb-specific to be here.

Reply:
Saving the OrientDB specific documentation here:
https://github.com/kensho-technologies/graphql-compiler/issues/930